code/controllers/subsystem/input.dm

```diff
@@ -34,6 +52,55 @@ SUBSYSTEM_DEF(input)
 		var/client/user = clients[i]
 		user.set_macros()
 
+///queue a click from usr onto clicked_atom with the given mouse handling arguments. only works if usr is the player mob clicking.
+/datum/controller/subsystem/input/proc/queue_click_from_usr(atom/clicked_atom, atom/location, control, params)
+	if(control != "mapwindow.map" || !ismob(usr) || QDELING(usr) || QDELETED(clicked_atom))
+		return FALSE
+
+	//high priority because clicks should be as low latency as possible, deferring to the beginning of the next SSinput run should rarely delay
+	//a click by more than a few milliseconds, but since the MC almost always resumes after every other sleeping proc this isnt guaranteed
+	if(!TICK_CHECK_HIGH_PRIORITY \
+		|| average_click_delay >= MAXIMUM_CLICK_LATENCY \
+		|| FOR_ADMINS_IF_CLICKS_BROKE_immediately_execute_all_clicks \
+		|| !initialized \
+		|| !(runlevels & Master.current_runlevel))
```

Comment:
Assertion, clicks from admins should never ever be delayed

Reply:
done